david_vallejos/space_challenge/src/main/java/com/tech_group/Simulation.java

```diff
@@ -0,0 +1,93 @@
+package main.java.com.tech_group;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+/**
+ * Class that contain Simulation methods.
+ */
+public class Simulation {
+
+    /**
+     * Loads items from txt file.
+     * @param fileName path followed with file name.
+     * @return a List of items from txt file.
+     */
+    public ArrayList<Item> loadItems(String fileName) {
+        ArrayList<Item> items = new ArrayList<>();
+
+        try {
+            ArrayList<String> fileList = (ArrayList<String>) Files.readAllLines(Paths.get(fileName), Charset.defaultCharset());
+            for (String item : fileList) {
+                items.add(new Item(item.split("=")[0], Integer.parseInt(item.split("=")[1])));
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        }
+        return items;
+    }
+
+    /**
+     * Loads U1 rocket with items that can carry.
+     * @param items that can be carry by rocket.
+     * @return a list of Rocket invested.
+     */
+    public ArrayList<Rocket> loadU1(ArrayList<Item> items) {
+        U1 rocket = new U1();
+        return getRocketList(rocket, items);
+    }
+
+    /**
+     * Gets rockets list that was filled.
+     * @param rocket object.
+     * @param items list that have been carry.
+     * @return a List of rocket filled out.
+     */
+    private ArrayList<Rocket> getRocketList(Rocket rocket, ArrayList<Item> items) {
```

Comment:
It looks like this method works, but it will be hard to maintain, imagine that instead of having only two types of rockets, we have six different models, maybe the solution could be nesting more if-else statements, but this would make hard understanding the code and maybe the implementation could cause bugs not easy to detect at first glance, maybe you can create a new method and delegates it the responsibility to determine which Rocket object to create.